Add toggle to suppress sending additional args to external players

trostboot opened the pull request:
# Title
<!-- Thanks for sending a pull request! Make sure to follow the contributing guidelines. -->
<!-- Important note, we may remove your pull request if you do not use this provided PR template correctly. -->
Add toggle to suppress sending additional args to external players

## Pull Request Type
<!-- Please select what type of pull request this is: [x] -->
- [ ] Bugfix
- [x] Feature Implementation
- [ ] Documentation
- [ ] Other

## Related issue
<!-- Please link the issue your pull request is referring to. -->
<!-- If this pull request fully resolves the relevant issue, put "closes" before the issue number. -->
<!-- Example: "closes #123456". -->
closes #4493 

## Description
<!-- Please write a clear and concise description of what the pull request does. -->
Adds a toggle setting in the external players sections that lets users suppress any args being sent to the external player aside from the video URL.
Also adds corresponding logic to skip processing of args if this toggle is enabled (this made the diff a mess even though it's a minimal change, tried different orders but git no likey).

## Screenshots <!-- If appropriate -->
<!-- Please add before and after screenshots if there is a visible change. -->
![ignoreArgs](https://github.com/FreeTubeApp/FreeTube/assets/17594477/5e45b4be-e4a6-40e5-a944-9ac1dda0e6b3)

## Testing <!-- for code that is not small enough to be easily understandable -->
<!-- Has this pull request been tested? -->
<!-- Please describe shortly how you tested it. -->
<!-- Are there any ramifications remaining? -->
Tested on Windows using SMPlayer and mpv with the toggle on and off. Seems to work as expected.
Testing with more combinations is always appreciated.
[Testing build](https://github.com/trostboot/FreeTube/actions/runs/7401284680).
## Desktop
<!-- Please complete the following information-->
- **OS:** Windows
- **OS Version:** 11 22H2
- **FreeTube version:** 0.19.1 

## Additional context
<!-- Add any other context about the pull request here. -->
I'm open to a better name for the setting if anyone thinks it's not clear enough.

PikachuEXE reviewed (approved):
I have no external player, can only review code

absidue commented:
Not sure about the naming, as it sounds like it will also ignore the users custom arguments, which is probably undesirable.

trostboot commented:
That's exactly what it does/what was requested in the linked issue. It also disables the custom args field if enabled to make that clear.

absidue commented:
If that is the case, then you should add a visual indicator to the custom arguments field, so that it is clear that it is disabled/ignored too. That being said, from a user perspective I would expect that to just disable passing any extra information. If I have gone out of my way to add custom arguments, then I would want them to be used.

trostboot commented:
It will grey out when disabled, sorry, I haven't updated the screenshot in the initial post, I added the logic to disable it afterwards.

I can see where you're coming from. What would you suggest, then? Pass on custom arguments anyway and rely on the user to clear the custom argument field if they don't them passed on for whatever reason, temporarily or otherwise?

trostboot commented:
Changed it so custom arguments are still passed on if there are any and changed the name and tooltip accordingly.

![ignoreArgs2](https://github.com/FreeTubeApp/FreeTube/assets/17594477/419ad957-3a9e-4355-9e39-ca46fed66b9e)

absidue reviewed (approved):
Hint for other reviewers: Turning on the "Hide Whitespace" option makes the diff significantly more readable.
